### ISSUE TITLE
Save and reuse private key, if none is specified

### ISSUE THREAD

**Issue by carver** [ethereum/trin#320]
That way, starting and re-running the node doesn't cycle your node ID. First mentioned in #15

**Comment by njgheorghita** [comment]
We do have the `--unsafe-private-key` flag which one can already use to avoid this. But, I guess you'd like to see this as the default behavior? There are instances when I do prefer random, cycled node IDs. Mostly when running quick & dirty tests with multiple local nodes running. Any thoughts as to how you could implement this in a way that still allows somebody to run multiple local nodes?

**Comment by carver** [comment]
> There are instances when I do prefer random, cycled node IDs. Mostly when running quick & dirty tests with multiple local nodes running. Any thoughts as to how you could implement this in a way that still allows somebody to run multiple local nodes?

Yeah, I liked `ddht`s solution of adding an `--ephemeral` flag, which did a bunch of things like cycling keys, using a temporary data directory, etc.

**Comment by carver** [comment]
Resolved by #621
